include/deal.II/grid/grid_in.h

```diff
@@ -567,6 +569,53 @@ class GridIn
               const double       tol                              = 1e-12,
               const bool         ignore_unsupported_element_types = true);
 
+  /**
+   * Read in a mesh stored in the ExodusII file format.
+   *
+   * ExodusII is a feature-rich file format that supports many more features
+   * (like node sets, finite element fields, quality assurance data, and more)
+   * than most other grid formats supported by this class. Many of these
+   * features do not have equivalent representations in deal.II and are
+   * therefore not supported (for example, deal.II does not assign degrees of
+   * freedom directly to nodes, so data stored in a nodal format is not loaded
+   * by this function). At the current time only the following information is
+   * extracted from the input file:
+   *
+   * <ol>
+   *   <li>Block ids: the block id of an element is loaded as its material
+   *     id.</li>
+   *   <li>Elements and vertices: the core geometric information stored in the
+   *     ExodusII file populates the attached Triangulation object. Higher-order
+   *     elements are automatically truncated to lower-order elements since
+   *     deal.II does not support this feature (e.g., there is no equivalent to
+   *     the <code>QUAD9</code> element in deal.II since all quadrilaterals have
+   *     four vertices and additional geometric information is either stored in
+   *     a Manifold or something like MappingQEulerian).</li>
+   *   <li>Sideset ids: these are interpreted as boundary ids (see the note on
+   *     @p boundary_id_to_sideset_ids below). An error will occur if you
+   *     attempt to read an ExodusII file that assigns a sideset id to an
+   *     internal face.</li>
+   * </ol>
+   *
+   * @param filename The name of the file to read from.
+   * @param boundary_id_to_sideset_ids A cell face in ExodusII can be in an
+   * arbitrary number of sidesets (i.e., it can have an arbitrary number of
+   * sideset ids) - however, a boundary cell face in deal.II has exactly one
+   * boundary id. Hence this function creates a mapping from deal.II boundary
+   * ids to the provided sideset ids. All boundary faces that are not in a
+   * sideset are given the (default) boundary id of $0$. This function then
+   * groups sidesets together into unique sets and gives each one a boundary id.
+   * For example: Consider a single-quadrilateral mesh whose left side has no
+   * sideset id, right side has sideset ids $0$ and $1$, and whose bottom and
+   * top sides have sideset ids of $0$. The left face will have a boundary id of
+   * $0$, the top and bottom faces boundary ids of $1$, and the right face a
+   * boundary id of $2$. Hence <code>boundary_id_to_sideset_ids</code> will be
+   * $\{\{\}, \{0\}, \{0, 1\}\}$.
+   */
+  void
+  read_exodusii(const std::string &            filename,
+                std::vector<std::vector<int>> &boundary_id_to_sideset_ids);
```

Comment:
Would it make sense to return this by value instead? A user can then ignore the argument.